pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/TabularParticipantParser.java

```diff
@@ -0,0 +1,329 @@
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.export.WorkflowAndFamilyIdExporter;
+import org.broadinstitute.dsm.model.Filter;
+import org.broadinstitute.dsm.model.ParticipantColumn;
+import org.broadinstitute.dsm.model.elastic.export.tabular.renderer.TextValueProvider;
+import org.broadinstitute.dsm.model.elastic.export.tabular.renderer.ValueProviderFactory;
+import org.broadinstitute.dsm.model.participant.ParticipantWrapperDto;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import org.broadinstitute.dsm.util.proxy.jackson.ObjectMapperSingleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Parses a list of ParticipantDtos from elasticSearch into a map of column names to string values.
+ * Uses a DDP instance to query the question schema to determine which questions are multi-select,
+ * and therefore may be spit into multiple columns
+ * Usage:
+ * p = new ParticipantDtoParser([[list of columns from UI]], DDPInstance)
+ * List configs = p.generateExportConfigs();
+ * List participantMaps = p.parse(listOfParticipantsFromES);
+ */
+public class TabularParticipantParser {
+    private static final Logger logger = LoggerFactory.getLogger(TabularParticipantParser.class);
+    private static final String COLUMN_SELECTED = "1";
+    private static final String COLUMN_UNSELECTED = "0";
+    private static final String DATA_AS_MAP = "dataAsMap";
+    private final List<Filter> filters;
+    private final DDPInstance ddpInstance;
+    private final boolean splitOptions;
+    private final boolean onlyMostRecent;
+    private final ValueProviderFactory valueProviderFactory = new ValueProviderFactory();
+
+    public TabularParticipantParser(List<Filter> filters, DDPInstance ddpInstance, boolean splitOptions, boolean onlyMostRecent) {
+        this.filters = filters;
+        this.ddpInstance = ddpInstance;
+        this.splitOptions = splitOptions;
+        this.onlyMostRecent = onlyMostRecent;
+    }
+
+    /**
+     * generates the config objects that will be used during parsing to put values into columns
+     * this makes a call to ElasticSearch to load question schemas, used to determine whether a given question
+     * is single or multiselect.
+     *
+     * @return the configs
+     */
+    public List<ModuleExportConfig> generateExportConfigs() {
+        List<ModuleExportConfig> configs = new ArrayList<>();
+        // map of table name => module export config
+        Map<String, ModuleExportConfig> exportInfoMap = new HashMap<>();
+        Map<String, Map<String, Object>> activityDefs = ElasticSearchUtil.getActivityDefinitions(ddpInstance);
```

Comment:
`ElasticSearchUtil` is an old class as well I would say, there are `ElasticSearchable` interface and `ElasticSearch` concrete implementation of `ElasticSearchable`, which I usually use for elastic search, `ElasticSearchable` may not have the function which is presented in `ElasticSearchUtil`, in this case I add new method to `ElasticSearchable`, if you plan to write unit tests for this class having `ElasticSearchUtil` directly here will struggle you I think.

Reply:
ah, that's helpful, I didn't know that ElasticSearchUtil was in the process of being deprecated.  I'll see if I can easily move over